src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2687,6 +2687,10 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                         Error(diagnostics, ErrorCode.ERR_NoImplicitConv, syntax, distinguisher.First, distinguisher.Second);
                     }
                 }
+                else if (sourceType.SpecialType == SpecialType.System_Void)
+                {
+                    Error(diagnostics, ErrorCode.ERR_NoImplicitConv, syntax, sourceType.ToDisplayString(), targetType.ToDisplayString());
```

Comment:
@aleksey Can you please provide a pointer to the "guidelines" you're referring to?